frontend/src/metabase/visualizations/components/ChartTooltip.jsx

```diff
@@ -19,7 +19,10 @@ export default class ChartTooltip extends Component {
     }
     if (Array.isArray(hovered.data)) {
       // Array of key, value, col: { data: [{ key, value, col }], element, event }
-      return hovered.data;
+      return hovered.data.map(d => ({
+        ...d,
+        key: d.key || getFriendlyName(d.col),
```

Comment:
The choropleth visualization wasn't setting `key` in each `data` item. I added a fallback in the `ChartTooltip` rather than patching the specific issue.

Separately, it'd be nice to document exactly what these click object data structures should contain.

Reply:
Why use `getFriendlyName` vs using `d.col.name`?

Reply:
I used `getFriendlyName` without thinking much about it since I saw that dimensions were using that in an adjacent code branch. It looks like `getFriendlyName` has logic to expand/translate aggregation names (e.g. `"avg"` => `` t`Average` ``) and falls back to `column.display_name`.

Reply:
Likely a rare edge case, but it seems like since `display_name` is user editable an admin could accidentally use the same `display_name` for two columns and cause a dup key problem here. Won't block on it, but figured it was worth noting.

Reply:
Hmm good point. We probably need to enforce uniqueness (within a given context) for all our display names.